File: src/main/java/net/golem/raknet/protocol/connected/connection/ConnectionRequestPacket.java
package net.golem.raknet.protocol.connected.connection;

import net.golem.raknet.codec.PacketDecoder;
import net.golem.raknet.codec.PacketEncoder;
import net.golem.raknet.protocol.DataPacket;
import net.golem.raknet.protocol.RakNetPacketIds;

public class ConnectionRequestPacket extends DataPacket {

	public long clientId;

	public long pingTime;

	public boolean useSecurity = false;

	public ConnectionRequestPacket() {
		super(RakNetPacketIds.CONNECTION_REQUEST);
	}

	@Override
	public void encode(PacketEncoder encoder) {
		encoder.writeLong(clientId);
		encoder.writeLong(pingTime);
		encoder.writeBoolean(useSecurity);
	}

	@Override
	public void decode(PacketDecoder decoder) {
		clientId = decoder.readLong();
		pingTime = decoder.readLong();
		useSecurity = decoder.readBoolean();
	}

	@Override
	public String toString() {
		return "ConnectionRequestPacket{" +
				"clientId=" + clientId +
				", pingTime=" + pingTime +
				", useSecurity=" + useSecurity +
				'}';
	}
}

File: src/main/kotlin/raknet/packet/protocol/connected/connection/ConnectionRequestPacket.kt
package raknet.packet.protocol.connected.connection

import io.netty.buffer.ByteBuf
import raknet.packet.DataPacket
import raknet.packet.PacketType

class ConnectionRequestPacket(
    var guid: Long,
    var time: Long,
): DataPacket(PacketType.CONNECTION_REQUEST.id()) {
    override fun encodeOrder(): Array<Any> {
        return arrayOf(guid, time)
    }

    companion object {
        fun from(data: ByteBuf): ConnectionRequestPacket {
            return ConnectionRequestPacket(
                data.readLong(),
                data.readLong()
            )
        }
    }

    override fun toString(): String {
        return "ConnectionRequestPacket(guid=$guid, time=$time)"
    }
}